app/Controllers/CoverController.php

```
@@ -155,6 +155,112 @@ public function download(Request $request, Response $response): Response
+            'domain' => $parsedUrl['host'] ?? ''
+        ]);
+
+        // Set upload directory (consistent with LibriController)
+        $uploadDir = $this->getCoversUploadPath() . '/';
+        if (!is_dir($uploadDir)) {
+            if (!mkdir($uploadDir, 0755, true)) {
+                throw new \RuntimeException(__('Impossibile creare la cartella di upload.'));
+            }
+        }
+
+        [$imageData, $effectiveUrl] = $this->downloadCover($initialUrl);
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$effectiveUrl'. (undefined)

(UnusedLocalVariable)